internal/restconfig/producer_test.go

```diff
@@ -0,0 +1,1031 @@
+
+			wasRun, err := t.producer.RunOnSelectedPrefixedContext(prefix,
+				func(clusterInfo *cluster.Info, namespace string, _ reporter.Interface) error {
+					t.actualProcessed++
+
+					Expect(clusterInfo.Name).To(Equal(cluster1Name))
+					Expect(namespace).To(Equal(prefixNS))
+
+					return nil
+				}, reporter.Stdout())
+
+			Expect(err).To(Succeed())
+			Expect(wasRun).To(BeTrue())
+		})
+
+		Context("and no prefix namespace is specified", func() {
+			It("should use the default prefix namespace", func() {
+				wasRun, err := t.producer.RunOnSelectedPrefixedContext(prefix,
+					func(clusterInfo *cluster.Info, namespace string, _ reporter.Interface) error {
+						t.actualProcessed++
+
+						Expect(clusterInfo.Name).To(Equal(cluster1Name))
+						Expect(namespace).To(Equal(defaultPrefixNS))
+
+						return nil
+					}, reporter.Stdout())
+
+				Expect(err).To(Succeed())
+				Expect(wasRun).To(BeTrue())
+			})
+		})
+	})
+
+	When("a default namespace is specified", func() {
+		BeforeEach(func() {
+			t.producer.WithDefaultNamespace(defaultNS)
+		})
+
+		It("should use the default namespace", func() {
+			wasRun, err := t.producer.RunOnSelectedPrefixedContext(prefix,
+				func(clusterInfo *cluster.Info, namespace string, _ reporter.Interface) error {
+					t.actualProcessed++
+
+					Expect(clusterInfo.Name).To(Equal(cluster1Name))
+					Expect(namespace).To(Equal(defaultNS))
+
+					return nil
+				}, reporter.Stdout())
+
+			Expect(err).To(Succeed())
+			Expect(wasRun).To(BeTrue())
+		})
+	})
+
+	When("the prefix kube config is specified", func() {
+		const prefixClusterName = "prefix-cluster"
+
+		var prefixConfigFile string
+
+		BeforeEach(func() {
+			clientConfig := api.NewConfig()
+			clientConfig.Clusters[prefixClusterName] = &api.Cluster{}
+			clientConfig.Contexts[prefixClusterName] = &api.Context{
+				Cluster: prefixClusterName,
+			}
+
+			prefixConfigFile = createKubeConfigFile(clientConfig)
+		})
+
+		JustBeforeEach(func() {
+			Expect(t.flags.Set(prefix+"context", prefixClusterName)).To(Succeed())
+			Expect(t.flags.Set(prefix+"config", prefixConfigFile)).To(Succeed())
+		})
+
+		It("", func() {
+			wasRun, err := t.producer.RunOnSelectedPrefixedContext(prefix,
+				func(clusterInfo *cluster.Info, namespace string, _ reporter.Interface) error {
+					t.actualProcessed++
+
+					Expect(clusterInfo.Name).To(Equal(prefixClusterName))
+					Expect(namespace).To(Equal(corev1.NamespaceDefault))
+
+					return nil
+				}, reporter.Stdout())
+
+			Expect(err).To(Succeed())
+			Expect(wasRun).To(BeTrue())
+		})
+	})
+
+	When("no prefix context is specified", func() {
+		JustBeforeEach(func() {
+			Expect(t.flags.Set(prefix+"context", "")).To(Succeed())
+			t.expectedProcessed = 0
+		})
+
+		It("should return false", func() {
+			wasRun, err := t.producer.RunOnSelectedPrefixedContext(prefix, func(_ *cluster.Info, _ string, _ reporter.Interface) error {
+				return nil
```

Comment:
```suggestion
				t.actualProcessed++
				return nil
```

Reply:
Was this supposed to be skipped?

Reply:
I assume skipped wrt using `noopPerContext` - no I missed thus one.